### ISSUE TITLE
Doesn't work with massive files

### ISSUE THREAD

**Issue by perkelle** [alefragnani/atom-language-pascal#26]
Syntax highlighting doesn't work with a large files. I have a pascal source file, which is really big (about 3 Mb) - highlighting doest not work with it

**Comment by alefragnani** [comment]
Hi @perkelle ,

That’s probably caused by the VS Code limitation, as reported in microsoft/vscode#3147 and updated in microsoft/vscode#6474 (comment). If the file is larger than the defined limit, VS Code simply does not activate the extension properly, and things like that happens.

If that is the case, there is nothing I can do in the extension to fix the issue. Sorry.

Hope this helps

**Comment by perkelle** [comment]
> Hi @perkelle ,
> 
> That’s probably caused by the VS Code limitation, as reported in [microsoft/vscode#3147](https://github.com/microsoft/vscode/issues/3147) and updated in [microsoft/vscode#6474](https://github.com/microsoft/vscode/issues/6474) (comment). If the file is larger than the defined limit, VS Code simply does not activate the extension properly, and things like that happens.
> 
> If that is the case, there is nothing I can do in the extension to fix the issue. Sorry.
> 
> Hope this helps

Does this apply to Atom editor for Mac?

**Comment by alefragnani** [comment]
Not that I’ aware of. 
But I remember Atom had “performance” issues while opening large files. Maybe the recent releases have fixed/improved, but I’m not sure.

**Comment by alefragnani** [comment]
Ops.. sorry! I totally missed you are talking about Atom, not VS Code. Ignore [that comment](https://github.com/alefragnani/atom-language-pascal/issues/26#issuecomment-511119070)

I don't have such a huge Pascal file to test right now, neither a Mac machine to test, but I'll try to see how it goes on Windows.

Again, sorry about the misunderstanding

**Comment by alefragnani** [comment]
I found some issues reported about file size in Atom Discuss and Atom repository. Probably an Atom limitation too:

- https://discuss.atom.io/t/maximum-file-size/46073
- https://discuss.atom.io/t/atom-file-size-restriction-is-it-still-set-at-2mb/43395/7
- https://github.com/atom/atom/issues/17120
- https://github.com/atom/atom/pull/14435

BTW, did you test using another syntax highlight plugin with that same 30Mb file?  You just have to open `Grammar Selector: Show` and choose another language, like C#, C++, Go, HTML.

Hope this helps

**Comment by perkelle** [comment]
My file is only 3Mb :) But I’ll try it, thanks

**Comment by perkelle** [comment]
I tried to use another syntax highlighting and it's work well for this file.

**Comment by alefragnani** [comment]
Hi @perkelle ,

I just found a valid 2.4Mb .pas file, and did reproduce the issue. 

I tested with `C#` and `Go` language packages (natives) and both also does not work. But, testing with `JavaScript` language package (also native), it worked. 

I'm not sure, but I guess the `JavaScript` package has been updated to use the new TreeSitter. The C#, did not yet.

If that is the case, I'm sorry to say but these files won't be supported _(at least for now)_, unless the Atom team develops some kind of _converter/migrator_ to TreeSitter parsers.

Hope this helps
